ast/src/parsed/mod.rs

```diff
@@ -43,12 +51,12 @@ pub enum PilStatement<T> {
 }
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
-pub struct SelectedExpressions<T, Ref = ShiftedPolynomialReference<T>> {
-    pub selector: Option<Expression<T, Ref>>,
-    pub expressions: Vec<Expression<T, Ref>>,
+pub struct SelectedExpressions<Expr> {
```

Comment:
More a general question, this doesn't really belong in the `parsed` module anymore right, because it now describes a concept that is used in all staged (with different concrete types), right?

Reply:
We could create a new module, but it's the same for most of the ast structs here, they are used across the stages.

Reply:
Also this module will get simpler, since I plan to remove ShiftedPolynomialReference, NamespacedPolynomialReference and IndexedPolynomialReference